internal/extensionsprovider/provider.go

```diff
@@ -47,38 +44,16 @@ type ExtensionsProvider func(irBytes []byte, groupID, conjureProject, version st
 //   - If the asset is a provider, invoke it with the relevant arguments to retrieve extension data.
 //   - Merge all extension maps from the assets into a single result.
 //   - Return the combined extensions map or an error if any asset invocation or JSON parsing fails.
-func New(configFile string, assets []string, url string) ExtensionsProvider {
+func NewAssetsExtensionsProvider(extensionsProviderAssets []string, configFile string, url string) ExtensionsProvider {
 	return func(irBytes []byte, groupID, conjureProject, version string) (map[string]any, error) {
 		irFile, err := tempfilecreator.WriteBytesToTempFile(irBytes)
 		if err != nil {
 			return nil, err
 		}
 
 		allExtensions := make(map[string]any)
-		for _, asset := range assets {
-			cmd := exec.Command(asset, "_assetInfo")
-			stdout, err := cmd.Output()
-			if err != nil {
-				if exitErr, ok := err.(*exec.ExitError); ok {
-					return nil, fmt.Errorf("%w: failed to execute %v\nstdout:\n%s\nstderr:\n%s", err, cmd.Args, string(stdout), string(exitErr.Stderr))
-				}
-				return nil, fmt.Errorf("%w: failed to execute %v\nstdout:\n%s", err, cmd.Args, string(stdout))
-			}
-
-			var response assetInfoResponse
-			if err := safejson.Unmarshal(stdout, &response); err != nil {
-				return nil, err
-			}
-
-			if response.Type == nil {
```

Comment:
Previous implementation was also doing a lot in this one method -- getting asset info, filtering based on it, constructing argument, invoking with argument, and accumulating, all in a single function.

I think the refactor makes this cleaner -- load of assets is handled as a precondition, getting the extensions from a particular asset is factored out, and then this function becomes much more targeted and just a matter of iterating over assets and accumulating.